### PR TITLE
Support Redis and Redis Cluster

### PR DESCRIPTION
**Pull request by JesusIslam**
Add support to Redis and Redis Cluster.

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @JesusIslam 

IMO, most of the codes (like DB API, TLS parsing) in redis and redis cluster are the same, I think we should combine them all. 

Maybe you can just provide a redis package and provide a type flag like `redis.type=[single|cluster]` to let you create different clients.

**Comment by siddontang** [comment]
Friendly Ping @JesusIslam

**Comment by JesusIslam** [comment]
@siddontang They are now combined under "redis" db.

File: db/redis/db.go [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+func getOptionsSingle(p *properties.Properties) *goredis.Options {
+	opts := &goredis.Options{}
+	opts.Network, _ = p.Get(redisNetwork)
+	opts.Addr, _ = p.Get(redisAddr)
+	opts.Password, _ = p.Get(redisPassword)
+	opts.DB = p.GetInt(redisDB, 0)
+	opts.MaxRetries = p.GetInt(redisMaxRetries, 0)
+
+	var err error
+	tmp, ok := p.Get(redisMinRetryBackoff)
```
Comment (siddontang) [inline db/redis/db.go]: why not using https://godoc.org/github.com/magiconair/properties#Properties.GetDuration directly?

Comment (JesusIslam) [reply db/redis/db.go]: Huh, you are right there are `GetDuration` and `GetParsedDuration`, must've been missed it.

**Comment by JesusIslam** [comment]
@siddontang alright it's done.

File: db/redis/db.go [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+			data = append(data, tmp)
+		}
+	}
+
+	return data, nil
+}
+
+func (r *redis) Update(ctx context.Context, table string, key string, values map[string][]byte) error {
+	data, err := json.Marshal(values)
```
Comment (siddontang) [inline db/redis/db.go]: you can't set this data directly, the update values only contain the fields need to be updated later, you need to get the old values, update them.

Comment (JesusIslam) [reply db/redis/db.go]: Ah, I didn't know that. I was assuming you give me the whole new value.

**Comment by JesusIslam** [comment]
@siddontang sorry for the mistake, it is now updated and should be work correctly.

File: db/redis/db.go (hunk 2 above)
Comment (JesusIslam) [reply db/redis/db.go]: Fixed in https://github.com/pingcap/go-ycsb/pull/89/commits/85a0affd7735da239a46d1b4d17428fa6f7ce7cc

**Pull request merged**